src/arsciexp.py

```diff
@@ -359,12 +407,46 @@ def MasterBias(self, fitsdict, det):
             return False
         else: # It must be the name of a file the user wishes to load
             msbias_name = self._argflag['run']['masterdir']+'/'+self._argflag['reduce']['usebias']
-            msbias = arload.load_master(msbias_name, frametype="bias")
+            msbias, head = arload.load_master(msbias_name, frametype="bias")
+            self._argflag['masters']['loaded'].append('bias')
         # Set and then delete the Master Bias frame
         self.SetMasterFrame(msbias, "bias", det)
         del msbias
         return True
 
+    def MasterRN(self, fitsdict, det):
+        """
+        Generate Master ReadNoise frame for a given detector
+
+        Parameters
+        ----------
+        fitsdict : dict
+          Contains relevant information from fits header files
+        det : int
+          Index of the detector
+
+        Returns
+        -------
+        boolean : bool
+          Should other ScienceExposure classes be updated?
+        """
+
+        # If the master bias is already made, use it
+        if self._msrn[det-1] is not None:
+            msgs.info("An identical master ReadNoise frame already exists")
+            return False
+        msrn = np.zeros((self._nspec[det-1], self._nspat[det-1]))
+        # Systems with multiple amps will need help here
+        if self._spect['det'][det-1]['numamplifiers'] > 1:
+            msgs.warn("Readnoise needs to be updated for multiple amps")
```

Comment:
Instead of a warning message, I would consider raising a bug, work, or error message.